harnessed_jobs/rtm_aliveness_power_on/v0/ccs_rtm_aliveness_power_on.py

```diff
@@ -0,0 +1,209 @@
+            self.__dict__[key] = CCS.attachSubsystem(value)
+
+ChannelInfo = namedtuple('ChannelInfo', ['reb_ps_channel', 'ts8_mon_chan',
+                                         'low_lim', 'high_lim', 'chkreb'])
+
+def set_monitoring_interval(ts8, period_ms, logger=logger):
+    logger.info("Setting tick and monitoring period to %i ms" % period_ms)
+    command = "change monitor-update taskPeriodMillis %i" % period_ms
+    logger.debug(command)
+    ts8.synchCommand(10, command)
+
+def power_off_rebs(lines=(0, 1, 2)):
+    # Power-off the requested REBs via the specified power-lines.
+    for power_line in lines:
+        command = "setNamedPowerOn %d master False" % power_line
+        logger.debug(command)
+        ccs_sub.rebps.synchCommand(10, command)
+
+def map_power_lines_to_rebs(ccs_sub, ntries=20, wait_between_tries=10,
+                            num_lines=3):
+    """
+    Map power lines to REBs by powering on one at a time for each REB
+    and trying to read the 1-wire ID register.  This function will
+    leave the REBs in a powered-off state.
+    """
+    rebids = ccs_sub.ts8.synchCommand(10, "getREBIds").getResult()
+    rebnames = ccs_sub.ts8.synchCommand(10, "getREBDeviceNames").getResult()
+
+    # Ensure that all of the power-lines to the REBs are off to start.
+    power_off_rebs()
+
+    # Loop over each REB to find the line it uses.
+    power_lines = {}
+    for rebid, rebname in zip(rebids, rebnames):
+        line = 0
+        power_line = None
+        while power_line is None and line < num_lines:
+            if line in power_lines.values():
+                line += 1
+                continue
+            for name in 'master digital analog'.split():
+                ccs_sub.rebps.synchCommand(10, 'setNamedPowerOn %d %s True'
+                                           % (line, name))
+                time.sleep(0.5)
+            for i in range(ntries):
+                logger.debug("%s, try %i", rebname, i)
+                try:
+                    ccs_sub.ts8.synchCommand(10, 'readRegister %s 1' % rebname)
+                    power_line = line
+                    logger.debug("%s: %i", rebname, power_line)
+                    break
+                except java.lang.Exception:
+                    time.sleep(wait_between_tries)
+            power_off_rebs(lines=(line,))
+            line += 1
+        if power_line is None:
+            raise java.lang.Exception("Could not read register of %s."
+                                      % rebname)
+        power_lines[rebid] = power_line
+    return power_lines
+
+def check_values(ccs_sub, rebid, name, rebps_channel, ts8_mon_chan, low_lim,
+                 high_lim, chkreb, logger=logger):
+    """
+    Check that power supply current (or voltage) levels are within the
+    specified range.
+    """
+    reb_channel_name = 'REB%d.%s' % (rebid, rebps_channel)
+    command = "getChannelValue %s" % reb_channel_name
+    logger.debug(command)
```

Comment:
@ohsaxton Should we use `getChannelValue` here or would `readChannelValue` be better?